Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/Lights/DirectionalLight.azsli

```diff
@@ -41,9 +41,10 @@ void ApplyDirectionalLights(Surface surface, inout LightingData lightingData)
         else if (o_transmission_mode == TransmissionMode::ThinObject) 
         {
             // Fetch and use shrinked positions for thin object transmission to ensure they fall onto the object when querying
+            // (surface.transmission.transmissionParams.x <-- shrink factor)
             DirectionalLightShadow::GetShadowCoords(
                 shadowIndex,
-                surface.position - lightingData.shrinkFactor * surface.vertexNormal,
+                surface.position - surface.transmission.transmissionParams.x * surface.vertexNormal,
```

Comment:
surface.transmission.transmissionParams.x isn't very readable, can you add functions to the transmission struct

float GetShrinkFactor() { return transmissionParams.x; }
float GetTransmissionNdLBias() { return transmissionParams.y; }
float GetDistanceAttenuation() { return transmissionParams.z; }

So that the code can read
surface.position - surface.transmission.GetShrinkFactor()
?

Reply:
Done! I've added these getters for thick mode transmission parameters too. 
Indeed, it is more readable now.